pkg/machines/reducers.es6

```diff
@@ -226,20 +226,34 @@ function storagePools(state, action) {
         const { connectionName, pools } = action.payload;
 
         const newState = Object.assign({}, state);
-        newState[connectionName] = {};
+        if (!(connectionName in newState))
+            newState[connectionName] = {};
+        else
+            newState[connectionName] = Object.assign({}, state[connectionName]);
 
-        // Array of strings (pool names)
-        pools.forEach(poolName => {
-            newState[connectionName][poolName] = []; // will be filled by UPDATE_STORAGE_VOLUMES
-        });
+        // Delete pools from state that are not in the payload
+        for (var poolCurrent in newState[connectionName]) {
+            if (!pools.includes(poolCurrent)) {
+                delete newState[connectionName][poolCurrent];
+            }
+        }
+
+        // Add new pools to state
+        for (var i in pools) {
+            let poolName = pools[i];
+            if (!(poolName in newState[connectionName])) {
+                newState[connectionName][poolName] = [];
+            }
+        }
```

Comment:
Is it possible that a pool is neither added nor removed, but changes properties? Then again the only information that we actually keep from a pool are its name and volumes, and the latter are initialized as an empty array as before (and still filled by `UPDATE_STORAGE_VOLUMES` I suppose?

(Sorry, more thinking aloud here, to understand what happens).

Reply:
So, if the volumes changed for some pool  (pool changes properties) the UPDATE_STORAGE_VOLUMES reducer should and will take care of it.
I tried to separate the logic between the two reducers, to not mess around with foreign state data.
UPDATE_STORAGE_POOLS should only add or remove storage pools but not check what are the volumes in these storage pools.
UPDATE_STORAGE_VOLUMES should only update storage volumes and not mess around with pools.
Hope that this makes sense.